app/scripts/directives/mobileServiceInstanceList.js

```diff
@@ -28,28 +26,78 @@
                               DataService,
                               Navigate) {
     var ctrl = this;
+    var serviceClassesVersion = APIService.getPreferredVersion('clusterserviceclasses');
+    var serviceInstanceVersion = APIService.getPreferredVersion('serviceinstances');
+    var serviceBindingsVersion = APIService.getPreferredVersion('servicebindings');
+    var isMobileService = $filter('isMobileService'); 
+    var isServiceInstanceReady = $filter('isServiceInstanceReady');
+    var isMobileClientEnabled = $filter('isMobileClientEnabled');
+    var isBindingReady = $filter('isBindingReady');
     ctrl.state = ctrl.state || {};
+    ctrl.boundInstances = [];
+    ctrl.unboundInstances = [];
     var watches = [];
 
+    ctrl.$onInit = function() {
+      ctrl.loaded = false;
+      DataService.list(serviceClassesVersion, {namespace: $routeParams.project}, function(serviceClasses) {
+        ctrl.serviceClasses = serviceClasses.by('metadata.name');
+        DataService.list(serviceInstanceVersion, {namespace: $routeParams.project}, function(serviceInstances) {
+          ctrl.serviceInstances = ctrl.filterClientBindableMobileServices(serviceInstances.by('metadata.name'));
+
+          DataService.list(serviceBindingsVersion, {namespace: $routeParams.project}, function(bindingsData) {
+            ctrl.bindings = bindingsData.by('metadata.name');
+            ctrl.separateBoundAndUnboundServices();
+            ctrl.loaded = true;
+          });
+        });
+      });
+
+      watches.push(DataService.watch(serviceInstanceVersion, {namespace: $routeParams.project}, function(serviceInstances) {
+        ctrl.serviceInstances = ctrl.filterClientBindableMobileServices(serviceInstances.by('metadata.name'));
+        ctrl.separateBoundAndUnboundServices();
+      }));
+
+      watches.push(DataService.watch(serviceBindingsVersion, {namespace: $routeParams.project}, function(bindingsData) {
+        ctrl.bindings = bindingsData.by('metadata.name');
+        ctrl.separateBoundAndUnboundServices();
+      }));
+    };
+
+    ctrl.filterClientBindableMobileServices = function(serviceInstances) {
+      if (!ctrl.serviceClasses) {
+        return;
+      }
+
+      return _.filter(serviceInstances, function(serviceInstance){
+        var serviceInstanceName = _.get(serviceInstance, 'spec.clusterServiceClassRef.name');
+        return isMobileService(ctrl.serviceClasses[serviceInstanceName]) && 
+               isServiceInstanceReady(serviceInstance) &&
+               isMobileClientEnabled(ctrl.serviceClasses[serviceInstanceName]);
+      });
+    };
+
+    ctrl.separateBoundAndUnboundServices = function() {
```

Comment:
Does this need to be added to the `ctrl`? It doesn't seem to be used in the view? Maybe have it as an internal function instead?